cmd/dreamboat/main.go

```diff
@@ -172,16 +198,105 @@ func setupKeys(c *cli.Context) (*blst.SecretKey, types.PublicKey, error) {
 
 func run() cli.ActionFunc {
 	return func(c *cli.Context) error {
-		g, ctx := errgroup.WithContext(c.Context)
+		if err := config.Validate(); err != nil {
+			return err
+		}
+
+		domainBuilder, err := pkg.ComputeDomain(types.DomainTypeAppBuilder, config.GenesisForkVersion, types.Root{}.String())
+		if err != nil {
+			return err
+		}
+
+		domainBeaconProposer, err := pkg.ComputeDomain(types.DomainTypeBeaconProposer, config.BellatrixForkVersion, config.GenesisValidatorsRoot)
+		if err != nil {
+			return err
+		}
 
-		// setup the relay service
-		service := &relay.DefaultService{
-			Log:    config.Log,
-			Config: config,
+		timeDataStoreStart := time.Now()
+		m := metrics.NewMetrics()
+
+		storage, err := badger.NewDatastore(config.Datadir, &badger.DefaultOptions)
+		if err != nil {
+			config.Log.WithError(err).Fatal("failed to initialize datastore")
+			return err
 		}
+		config.Log.
+			WithFields(logrus.Fields{
+				"service":     "datastore",
+				"startTimeMs": time.Since(timeDataStoreStart).Milliseconds(),
+			}).Info("data store initialized")
+
+		timeRelayStart := time.Now()
+
+		as := &pkg.AtomicState{}
 
+		ds := datastore.NewDatastore(&datastore.TTLDatastoreBatcher{storage}, storage.DB)
+		if err = datastore.InitDatastoreMetrics(m); err != nil {
+			return err
+		}
+
+		regMgr := relay.NewProcessManager(c.Uint("relay-verify-queue-size"), c.Uint("relay-store-queue-size"))
+		regMgr.AttachMetrics(m)
+
+		reg, err := ds.GetAllRegistration()
```

Comment:
Could we extract this into a method to make the `run` call more readable?

Reply:
To function of main package? like setupKeys above? Or where would you like to see it?

Reply:
Yes, nice! But could we make it a method of `ProcessManager`?

Reply:
For processmanager - definitely not - as it doesn't really include the database. 
I can add this into the relay itself, however it may not necessarily be relay's area of interest, and separated from it. 
This one is only related to the process of initially loading this particular binary, that's why it's placed in the cmd folder and in neither of other modules.